org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/XMLLanguageServer.java

```diff
@@ -135,6 +143,22 @@ public void updateSettings(Object initializationOptionsSettings) {
 			if (newCompletions != null) {
 				xmlTextDocumentService.updateCompletionSettings(newCompletions);
 			}
+
+			ServerSettings serverSettings = xmlClientSettings.getServer();
+			if(serverSettings != null) {
+				String workDir = serverSettings.getWorkDir();
+				workDir = FilesUtils.validatePath(workDir);
+				if(workDir != null) {
+					if((workDir).equals(System.getProperty(FilesUtils.LSP4XML_WORKDIR_KEY)) == false) {
+						System.setProperty(FilesUtils.LSP4XML_WORKDIR_KEY, workDir);

```

Comment:
no no no. You don't set system props within your program (unless it's limited in scope, usually during tests)